hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientMapProxy.java

```diff
@@ -698,11 +699,12 @@ public void loadAll(Set<K> keys, boolean replaceExistingValues) {
         MapKeySetRequest request = new MapKeySetRequest(name);
         MapKeySet mapKeySet = invoke(request);
         Set<Data> keySetData = mapKeySet.getKeySet();
-        Set<K> keySet = new HashSet<K>(keySetData.size());
+        InflatableSet<K> keySet = new InflatableSet<K>(keySetData.size());
         for (Data data : keySetData) {
             final K key = toObject(data);
```

Comment:
a better better expressed objections against introducing dependency on SS: Mixing concerns. Responsibility of this class is to inflate on modification/lookup. I don't want to deal with de-serialization in the same class - it has a nice responsibility right now. However I have an idea, stay tuned :)